correctness/mockmetheus.py

```diff
@@ -0,0 +1,289 @@
+import os
+import re
+import time
+import requests
+import snappy
+from prom_pb2 import (
+    ReadRequest,
+    Query,
+    ReadResponse,
+    LabelMatcher,
+    WriteRequest,
+    TimeSeries,
+    Label,
+    Sample,
+)
+from google.protobuf.json_format import MessageToDict
+
+"""
+Mockmetheus is a helper class that mocks remote operations from Prometheus.
+It uses the same protobuf definitions as Prometheus to construct snappy-encoded
+payloads for both remote-read and remote-write requests and responses.
+"""
+class Mockmetheus:
```

Comment:
What is the motivation for using Python rather than Go. We are doing a lot of regex and custom parsing here for Prometheus requests rather than using Prometheus libraries.

For example we can use the `github.com/prometheus/prometheus/promql/parser` library for parsing Prometheus query strings. Using Prometheus libraries would also have the added benefit of being able to upgrade the dependencies for functional changes / bugs, and not mixing Go and Python without warrant.

Reply:
Although we can use any language for this implementation, we need some reason for doing so. If there is no benefit or clear purpose here for using Python, I would prefer we implement in Go. There is no need to write the parsing logic when we have the native parsing libraries available. Here is a small Go program which can parse a Prometheus query string, rather than introducing the required logic ourselves:

```go
package main

import (
	"fmt"
	"log"

	"github.com/prometheus/prometheus/promql/parser"
)

func main() {
	query := `scrape_duration_seconds{}[5d]`
	node, err := parser.ParseExpr(query)
	if err != nil {
		log.Fatalf("Error parsing query: %v", err)
	}
	PrintQuery(node)
}

func PrintQuery(node parser.Node) {
	switch n := node.(type) {
	case *parser.MatrixSelector:
    fmt.Println("Metric name:", n.VectorSelector)
		fmt.Println("Range:", n.Range)
	default:
		fmt.Printf("Unknown query node type: %T\n", n)
	}
}
```